prov/efa/src/efa_user_info.c

```diff
@@ -269,34 +269,125 @@ int efa_user_info_get_dgram(uint32_t version, const char *node, const char *serv
 	return -FI_ENODATA;
 }
 
+#if HAVE_CUDA || HAVE_NEURON || HAVE_SYNAPSEAI
+/**
+ * @brief determine if EFA provider should claim support of FI_HMEM in info
+ * @param[in]	version		libfabric API version used by user
+ * @param[in]	hints		hints from user's call to fi_getinfo()
+ * @return	true, if EFA provider should claim support of FI_HMEM
+ * 		false, otherwise
+ */
+bool efa_user_info_should_support_hmem(int version, const struct fi_info *hints)
+{
+	/*
+	 * FI_HMEM is a primary capability. If user did not explicitly request
+	 * it, we should not claim support of it.
+	 */
+	if (!hints || !(hints->caps & FI_HMEM))
+		return false;
+
+	/* Note that the default behavior of EFA provider is different between
+	 * libfabric API version.
+	 *
+	 * For libfabric API version 1.17 and earlier, EFA provider does not
+	 * support FI_HMEM unless GPUDirect RDMA is available.
```

Comment:
Do we need to elevate some of these comments to publicly visible documentation somewhere?  It seems problematic to me that the user might read up-to-date documentation about 1.18 but not realize that their initialization uses 1.16 and get confused as to why FI_HMEM is not being supported.

Reply:
> Do we need to elevate some of these comments to publicly visible documentation somewhere? It seems problematic to me that the user might read up-to-date documentation about 1.18 but not realize that their initialization uses 1.16 and get confused as to why FI_HMEM is not being supported.

Just got hit by this gotcha. Would be nice to see some extra documentation.